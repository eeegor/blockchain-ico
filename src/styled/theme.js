/**
 *
 * Theme
 *
 * INFO: All size values in this theme are px
 */
export const colors = {
	primaryDarkBlue: '#061258',
	primaryLightBlue: '#3866FF',
	primaryPurple: '#6F3BFF',
	primaryTeal: '#06CFC5',
	primaryGreen: '#3BE3AE',
	primaryRed: '#F54269',
	primaryText: '#101530',
	secondaryText: '#33394A',
	text: '#5D616F',
	lightText: '#A3A6AD',
	secondaryBg: '#18223C',
	primaryBg: '#33394A',
	platformBg: '#EAEBED',
	websiteBg: '#F6F7FB',
	divider: '#EAEBED',
	bitcoin: '#F7931A',
	litecoin: '#88CBF5',
	etherium: '#5F88A8',
	buttonPrimary: '#2B60FF',
	buttonPrimaryBg: '#3870FF'
};

export const fonts = {
	gtCinetype: 'GTCinetype-Regular',
	gtCinetypeItalic: 'GTCinetype-Italic',
	gtCinetypeBold: 'GTCinetype-Bold',
	gtCinetypeBoldItalic: 'GTCinetype-Bold-Italic',
	gtCinetypeLight: 'GTCinetype-Light',
	gtCinetypeLightItalic: 'GTCinetype-Light-Italic',
	gtCinetypeMono: 'GTCinetype-Mono',
	rubik: 'Rubik-Regular',
	rubikMedium: 'Rubik-Medium'
};

export const theme = {
	colors,
	fonts,
	fontSizeDefault: [16, 18],
	borderRadius: 4,
	container: {
		maxWidth: 1030
	},
	grid: {
		maxWidth: 1030,
		gridGap: 30
	},
	space: [0, 4, 8, 16, 32, 64, 128, 256],
	shadows: {
		small: '0px 2px 3px 0px rgba(0, 14, 70, .1)',
		medium: '2px 4px 5px rgba(0, 14, 70, .17)',
		large: '4px 12px 24px rgba(0, 14, 70, .11)',
		button: '0 14px 17px rgba(0, 4, 18, .16)'
	},
	media: {
		xSmall: 320,
		small: 480,
		tablet: 767,
		desktop: 1024,
		large: 1280,
		xLarge: 1440
	},
	headerHeight: {
		mobile: 60,
		tablet: 84
	}
};
