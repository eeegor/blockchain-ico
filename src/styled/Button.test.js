import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Button } from '.';
import { theme } from './theme';

describe('Button', () => {
	it('renders button', () => {
		const tree = renderer.create(<Button />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', theme.colors.buttonPrimary);
		expect(tree).toHaveStyleRule('background', '#fff');
	});

	it('renders selected button', () => {
		const tree = renderer.create(<Button selected />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', '#fff');
		expect(tree).toHaveStyleRule('background', theme.colors.buttonPrimary);
	});

	it('renders outline button', () => {
		const tree = renderer.create(<Button outline />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', theme.colors.buttonPrimary);
		expect(tree).toHaveStyleRule('background', 'none');
	});

	it('renders primary button', () => {
		const tree = renderer.create(<Button primary />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', '#fff');
		expect(tree).toHaveStyleRule('background', theme.colors.buttonPrimary);
	});

	it('renders success button', () => {
		const tree = renderer.create(<Button success />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', '#fff');
		expect(tree).toHaveStyleRule('background', theme.colors.primaryGreen);
	});

	it('renders danger button', () => {
		const tree = renderer.create(<Button danger />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', '#fff');
		expect(tree).toHaveStyleRule('background', theme.colors.primaryRed);
	});

	it('renders button with no shadow', () => {
		const tree = renderer.create(<Button noShadow />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('box-shadow', 'none !important');
	});

	it('renders token button', () => {
		const tree = renderer.create(<Button token />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', '#fff');
		expect(tree).toHaveStyleRule(
			'background',
			'linear-gradient(45deg,#6947FF 0%,#602BFF 100%)'
		);
	});
});
