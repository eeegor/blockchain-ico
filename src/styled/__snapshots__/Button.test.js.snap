// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

<button
  className="c0"
/>
`;

exports[`Button renders button with no shadow 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  box-shadow: none !important;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

<button
  className="c0"
/>
`;

exports[`Button renders danger button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  color: #fff;
  background: #F54269;
  border: 1px solid #F54269;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c0:hover {
  color: #fff;
  background: #f21243;
  border-color: #f21243;
}

<button
  className="c0"
/>
`;

exports[`Button renders outline button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  color: #2B60FF;
  background: none;
  border: 1px solid #2B60FF;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

<button
  className="c0"
/>
`;

exports[`Button renders primary button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  color: #fff;
  background: #2B60FF;
  border: 1px solid #2B60FF;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

<button
  className="c0"
/>
`;

exports[`Button renders selected button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  color: #fff;
  background: #2B60FF;
  border: 1px solid #2B60FF;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

<button
  className="c0"
  selected={true}
/>
`;

exports[`Button renders success button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  color: #fff;
  background: #3BE3AE;
  border: 1px solid #3BE3AE;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c0:hover {
  color: #fff;
  background: #1dce96;
  border-color: #1dce96;
}

<button
  className="c0"
/>
`;

exports[`Button renders token button 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  color: #fff;
  border: 1px solid #6947FF;
  background: #6947FF;
  background: linear-gradient(45deg,#6947FF 0%,#602BFF 100%);
  position: relative;
  z-index: 2;
}

.c0:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c0:before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  content: '';
  background: #000;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  opacity: 0;
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  z-index: -1;
}

.c0:hover {
  color: #fff;
  border: 1px solid #6947FF;
  background: #6947FF;
  background: linear-gradient(45deg,#6947FF 0%,#602BFF 100%);
}

.c0:hover:before {
  opacity: 0.2;
}

<button
  className="c0"
/>
`;
